src/Pos/Update/Poll/PollState.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+-- Types related Poll update system
+
+module Pos.Update.Poll.PollState
+       (
+         -- * Types
+         PollState (..)
+
+       -- * Lenses
+       -- * PollState
+       , psBlockVersions
+       , psAdoptedBV
+       , psSoftwareVersion
+       , psEpochProposers
+       , psConfirmedBVs
+       , psConfirmedProposals
+       , psActiveProposals
+       , psDelActivePropsIdx
+       , psSlottingData
+       , psUndecidedProposals
+       , psDecidedProposals
+       , psStakePerEpoch
+       , psMultiRichmenStake
+       , psIssuersStakes
+       ) where
+
+import           Control.Lens                 (makeLenses)
+import           Data.HashSet                 (HashSet)
+import           Data.HashMap.Strict          (HashMap)
+import           Universum
+
+import           Pos.Core.Types               (BlockVersionData, Coin)
+import           Pos.Lrc.DB.Issuers           (IssuersStakes)
+import           Pos.Slotting.Types           (SlottingData)
+import           Pos.Ssc.GodTossing.Toss.Pure (MultiRichmenStake)
+import           Pos.Types                    (ApplicationName, BlockVersion,
+                                               EpochIndex, NumSoftwareVersion,
+                                               SoftwareVersion, StakeholderId)
+import           Pos.Update.Core              (UpId)
+import           Pos.Update.Poll              (BlockVersionState,
+                                               ConfirmedProposalState,
+                                               DecidedProposalState, ProposalState,
+                                               UndecidedProposalState)
+import           Pos.Util.Modifier            (MapModifier)
+
+data PollState = PollState
+    { -- | All proposed block versions with their states
+      _psBlockVersions      :: !(MapModifier BlockVersion BlockVersionState)
+      -- | Presently adopted block version with its data
+    , _psAdoptedBV          :: !(BlockVersion, BlockVersionData)
+      -- | Software version of this node
+    , _psSoftwareVersion    :: !SoftwareVersion
```

Comment:
This and other things you pointed out were the result of the unnecessary `MonadPollRead` instance. Removing it addresses your concerns.